lib/DOMFetcher.js

```diff
@@ -0,0 +1,62 @@
+/* globals Services */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/Services.jsm");
+
+/**
+ * An event listener that will capture the DOM
+ * content of every closing HTMLDocument.
+ */
+function DOMFetcher() {
+  this.listeners = new Set();
+  Services.mm.loadFrameScript("resource://loci/data/content/DocumentChangeMessenger.js", true);
+  this._messageListener = this._messageListener.bind(this);
+  Services.mm.addMessageListener("loci@document-unloading", this._messageListener, true);
```

Comment:
as mentioned above, one listener will make things much simpler later on, when we have more messages